lib/pdftk/pdf.rb

```diff
@@ -41,19 +40,29 @@ def xfdf
 
     def fields
       unless @_all_fields
-        field_output = `pdftk "#{path}" dump_data_fields`
-        raw_fields   = field_output.split(/^---\n/).reject {|text| text.empty? }
+        field_output = run_cmd path, "dump_data_fields"
+        raw_fields = field_output.split(/^---\n/).reject { |text| text.empty? }
         @_all_fields = raw_fields.map do |field_text|
           attributes = {}
           field_text.scan(/^(\w+): (.*)$/) do |key, value|
             attributes[key] = value
           end
           Field.new(attributes)
         end
-        @_fields_mapping = Hash[@_all_fields.each_with_index.map{|field, index| [field.name,index]}]
+        @_fields_mapping = @_all_fields.each_with_index.map { |field, index| [field.name, index] }.to_h
```

Comment:
Maybe we can cut this down a bit?
```suggestion
        @_fields_mapping = @_all_fields.each_with_index.to_h { |field, index| [field.name, index] }
```